server/handler/base.go

```diff
@@ -46,9 +46,17 @@ type Base struct {
 }
 
 type PullEvaluationOptions struct {
-	PolicyPath         string `yaml:"policy_path"`
+	AppName    string `yaml:"app_name"`
+	PolicyPath string `yaml:"policy_path"`
+
+	// StatusCheckContext will be used to create the status context. It will be used in the following
+	// pattern: <StatusCheckContext>: <Base Branch Name>
 	StatusCheckContext string `yaml:"status_check_context"`
-	AppName            string `yaml:"app_name"`
+
+	// StatusCheckLegacyContext enables the sending of a second status using just StatusCheckContext as the context,
+	// no templating. This is turned off by default. This is to support legacy workflows that depend on the original
+	// context behaviour, and will be removed in 2.0
+	StatusCheckLegacyContext bool `yaml:"status_check_legacy_context"`
```

Comment:
I think it would be better to name this more like an option. The current name makes it seem like it should be a string.

```suggestion
	PostInsecureStatusChecks bool `yaml:"post_insecure_status_checks"`
```